packages/react-app/src/App.jsx

```diff
@@ -855,38 +826,72 @@ function App(props) {
   const [toAddress, setToAddress] = useLocalStorage("punkWalletToAddress", "", 120000);
 
   const [amount, setAmount] = useState();
+  console.log("amount", amount);
 
   const [amountEthMode, setAmountEthMode] = useState(false);
 
   const [receiveMode, setReceiveMode] = useState(false);
 
-  if (window.location.pathname !== "/") {
-    try {
-      const path = window.location.pathname.replace("/", "");
+  // ERC20 Token balance to use in ERC20Balance and in ERC20Input
+  const [balanceERC20, setBalanceERC20] = useState(null);
 
-      if (path.startsWith("ethereum:")) {
-        const eip681URL = window.location.href.substring(window.location.href.indexOf("ethereum:"));
+  const [priceERC20, setPriceERC20] = useState();
+  console.log("priceERC20", priceERC20);
+
+  const switchToTokenAddress = localStorage.getItem("switchToTokenAddress");
+
+  if (switchToTokenAddress) {
+    localStorage.removeItem("switchToTokenAddress");
+    const storedAmount = localStorage.getItem("amount");
+    if (storedAmount) {
+      localStorage.removeItem("amount");
+    }
 
-        const eip681Object = parse(eip681URL);
-        console.log("eip681Object", eip681Object);
+    let tokens = targetNetwork?.erc20Tokens;
 
-        const chainId = eip681Object.chain_id;
+    if (tokens) {
+      const customTokens = tokenSettingsHelper.getCustomItems();
 
-        handleNetworkByQR(chainId, networkSettingsHelper, setTargetNetwork)
+      if (customTokens.length > 0) {
+        tokens = tokens.concat(customTokens);
+      }
 
-        const functionName = eip681Object.function_name;
-        const tokenAddress = eip681Object?.target_address;
+      const token = tokens.find(token => token.address.toLowerCase() === switchToTokenAddress.toLowerCase());
 
-        let toAddress;
+      if (token) {
+        setPriceERC20(null);
 
-        if (functionName == "transfer" && tokenAddress) {
-          localStorage.setItem("switchToTokenAddress", tokenAddress);
-          toAddress = eip681Object?.params?.address;
+        if (selectedErc20Token?.address.toLowerCase() !== switchToTokenAddress.toLowerCase()) {
+          tokenSettingsHelper.updateSelectedName(token.name);
         }
-        else {
-          localStorage.setItem("switchToEth", true);
-          toAddress = eip681Object?.target_address
+
+        if (storedAmount) {
+          const amountBigNumber = BigNumber.from(storedAmount);
+          setAmount(amountBigNumber);
         }
+      } else {
+        showModal(TOKEN_ERROR.NOT_SUPPORTED + " :" + switchToTokenAddress);
```

Comment:
the error message is wrong I think, token error shouldn't use `switchToTokenAddress`